pkg/syncer/specsyncer.go

```diff
@@ -86,11 +86,13 @@ func NewSpecSyncer(from, to *rest.Config, syncedResourceTypes []string, kcpClust
 	return New(kcpClusterName, pclusterID, fromDiscovery, fromClient, toClient, KcpToPhysicalCluster, syncedResourceTypes, pclusterID)
 }
 
-func (c *Controller) deleteFromDownstream(ctx context.Context, gvr schema.GroupVersionResource, namespace, name string) error {
-	// TODO: get UID of just-deleted object and pass it as a precondition on this delete.
-	// This would avoid races where an object is deleted and another object with the same name is created immediately after.
-
-	return c.toClient.Resource(gvr).Namespace(namespace).Delete(ctx, name, metav1.DeleteOptions{})
+func (c *Controller) deleteFromDownstream(ctx context.Context, gvr schema.GroupVersionResource, upstreamNamespace, name string, downstreamUID string) error {
+	if err := c.toClient.Resource(gvr).Namespace(upstreamNamespace).Delete(ctx, name, metav1.DeleteOptions{
+		Preconditions: metav1.NewUIDPreconditions(downstreamUID),
+	}); err != nil && !k8serrors.IsNotFound(err) && !k8serrors.IsConflict(err) {
```

Comment:
What would `k8serrors.IsConflict` being true indicate, and why is it desirable to silently ignore that?

Reply:
In that particular case, that is when the UID set in the pre-condition differs from the UID of the downstream object that actually exists, a conflict implies the object that was meant to be deleted no longer exists, and has been replaced with a new object, hence the operation that is being performed has already been fulfilled, and the error can be ignored, similarly to the 404 case.